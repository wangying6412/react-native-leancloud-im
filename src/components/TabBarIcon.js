/**
 * TabBar的图标
 *
 * @author lihaitang
 * @version v0.0.1
 */

import React from 'react';

import PropTypes from 'prop-types';

import Icon from 'react-native-vector-icons/Ionicons';

class TabBarIcon extends React.Component{

    static get propTypes(){
        return {
            icon : PropTypes.string.isRequired,
            focused : PropTypes.bool.isRequired,
            tintColor : PropTypes.string.isRequired,
        };
    }

    render(){
        let {
            icon,
            focused,
            tintColor,
        } = this.props;

        icon = focused ? icon : /\-outline/.test(icon) ? icon : icon+'-outline';
        let style = !focused ? {} : { color : tintColor };
        return(
            <Icon name={icon} size={26} style={style} />
        );
    }
}

export default TabBarIcon;




