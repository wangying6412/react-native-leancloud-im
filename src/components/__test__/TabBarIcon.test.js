/* global describe test expect */

import React from  'react';
import 'react-native';
import renderer from 'react-test-renderer';

import TabBarIcon from '../TabBarIcon';

describe('TabBarIcon',()=>{
    test('TabBarIcon的快照测试',()=>{
        const component = renderer.create(
            <TabBarIcon icon="ios-home" focused={false} tintColor="#F00" />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
