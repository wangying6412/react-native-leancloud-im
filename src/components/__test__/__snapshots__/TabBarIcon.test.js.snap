// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarIcon TabBarIcon的快照测试 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 26,
      },
      Object {},
      Object {
        "fontFamily": "Ionicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;
